File: MEET.cpp
#include "iostream"
// #include  <string>
using namespace std;
std::string convert24(string time)
{
    char part = time[6];
    // cout<<part<<endl;
    if (part=='P')
    {
        string hour = time.substr(0,2);
        // cout<<hour<<endl;
        int hr = stoi(hour);
        // cout<<hr<<endl;
        if (hr==12)
        {
            hr=0;
        }
        
        hr+=12;
        string minutes = time.substr(3,2);
        string result = to_string(hr).append(minutes);
        return result;
    }
    else if (part=='A')
    {
        string hour = time.substr(0,2);
        // cout<<hour<<endl;
        int hr = stoi(hour);
        if (hr==12)
        {
            return "0000";
        }
        
        // cout<<hr<<endl;
        string minutes = time.substr(3,2);
        string result = to_string(hr).append(minutes);
        if (hr<10)
        {
            result.insert(0,"0");
        }
        return result;
    }
    
    
    return 0;
}
int main()
{
    string t;
    getline(cin,t);//test case input
    int tests = stoi(t);
    for(int i=0;i<tests;i++)
    {
        string time = "02:00 PM";
        getline(cin,time);// my time input
        // cout<<time<<endl;
        // cout<<convert24(time)<<endl;
        std::string time24 = convert24(time);
        int arr[4];
        string n;
        cin >> n;// n input
        unsigned int N = stoi(n);
        for (int j = 0; j < N+1; j++)
        {
            
            string LRTIME;
            getline(cin,LRTIME);// lr time input
            string delim = "M";
            // cout<<LRTIME.find(delim)<<endl;

            string ltime = LRTIME.substr(0,LRTIME.find(delim)+1);
            string rtime = LRTIME.substr(LRTIME.find(delim)+1);
            // cout<<ltime<<"\t\t"<<rtime<<endl;
            string ltime24 = convert24(ltime);
            string rtime24 = convert24(rtime);
            cout<<ltime24<<"\t\t"<<rtime24<<endl;
        }
        
    }
    
    
    
    
    return 1;
}

File: MEET.py
def convert24(time):
    result=""
    if time[6] == 'A':
        hour = time[0:2]
        minutes=time[3:5]
        hr = int(hour)
        if(hr==12):
            return "00"+minutes
        if hr < 10:
            return "0"+str(hr)+minutes

        result =  str(hr)+minutes
    if time[6] == 'P':
        hour = time[0:2]
        hr = int(hour)
        if hr==12:
            hr=0
        hr+=12
        minutes=time[3:5]
        # print()
        result = str(hr)+str(minutes)
    # print(result)
    return result

# print(convert24("10:40 AM"))
for t in range(int(input())):
    mytime=input()
    # print(convert24(mytime))
    n = int(input())
    for i in range(n):
        LRTIME = input()
        # print(LRTIME[0:8])
        ltime = LRTIME[0:8]
        rtime = LRTIME[9:]
        # print(ltime)
        # print(rtime)
        ltime24 = int(convert24(ltime))
        rtime24 = int(convert24(rtime))
        # print(ltime24)
        # print(rtime24)
        mytime24 = int(convert24(mytime))
        # print(mytime24)
        if ltime24<=mytime24 and mytime24<=rtime24:
            print('1',end="")
        else:
            print("0",end="")
    print("\n")